### DISCUSSION TITLE
Load the UI when hovering over an object of a certain type, such as InteractableVehicle or InteractableStorage

### DISCUSSION THREAD

**Discussion by barashF** [SmartlyDressedGames/Unturned-3.x-Community#3999] [Q&A]
I tried to implement it through UnturnedPlayerEvents.PlayerGesture.None, but as it turned out it doesn’t work like that, I thought to implement it through a timer, but I’m afraid of an extra load on the server, I will be grateful if you tell me an alternative implementation.

**Comment by GazziFX** [comment]
This should not create high load when at 50 players, if you want smoother frametime you can spread loop into several frames
```csharp
private float lastUpdate;

void Update()
{
   float time = Time.time;
   if (time - lastUpdate < 0.1f)
      return;

   foreach (var client in Provider.clients)
   {
      if (Physics.Raycast(...))
         // check components and activate ui
   }
}
```
